README.md

```diff
@@ -1,123 +1,323 @@
 ```
-gradle spotlessApply
+2. Create a gcs file with the query:
 
-gradle build
+```
+export REID_QUERY_BUCKET=<name>
+cat << EOF | gsutil cp - gs://${REID_QUERY_BUCKET}/reid_query.sql
+${QUERY}
+EOF
+```
+3. Create a pub-sub topic. Refer [create a topic](https://cloud.google.com/pubsub/docs/create-topic#create_a_topic)
+
+4. Run the pipeline by passing required parameters:
+```
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 
+-Pargs="--region=<region> 
+--project=<project_id>
+--streaming --enableStreamingEngine 
+--tempLocation=gs://<bucket>/temp 
+--numWorkers=5 --maxNumWorkers=10 
+--runner=DataflowRunner 
+--tableRef=<project_id>:<dataset>.<table> 
+--dataset=<dataset> 
+--topic=projects/<project_id>/topics/<name> 
+--autoscalingAlgorithm=THROUGHPUT_BASED 
+--workerMachineType=n1-highmem-4 
+--deidentifyTemplateName=projects/<project_id>/deidentifyTemplates/<name> 
+--DLPMethod=REID 
+--keyRange=1024 
+--queryPath=gs://${REID_QUERY_BUCKET}/reid_query.sql"
 
-// inspect is default as DLP Method; For deid: --DLPMethod=DEID
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=s3://<bucket>>/file.csv --dataset=<name>  --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --awsRegion=<aws_region> --awsCredentialsProvider=$AWS_CRED"
 ```
 
-## DeIdentification of JSONL files
-The pipeline supports JSONL file format where each line is a valid JSON Object and newline character is used to separate JSON objects. A sample file can be found in [test resources](src/test/resources/CCRecords_sample.jsonl).
+### Pipeline Parameters
+
+Following pipeline options have 
+
+| Pipeline Option                  | Description                                                                                                                                                                                                                                                        | Used in Operations  |
+|----------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------|
+| `region`                         | Specifies a regional endpoint for deploying your Dataflow jobs.                                                                                                                                                                                                    | All                 |
```

Comment:
Is there any limitations to the region that can be provided? Like if my bucket is in some other region, can I run my job as global or any other region?

Reply:
We need to check on this. will investigate and get back. However, this can be added in "How to adapt pipeline to your use case section" and not in the pipeline options section.

Reply:
Added https://github.com/GoogleCloudPlatform/dlp-dataflow-deidentification/blob/readme-revamp/README.md#how-to-adapt-this-pipeline-for-your-use-cases